docs/en/ingest-management/fleet-agent-proxy-support.asciidoc

```diff
@@ -0,0 +1,310 @@
+[[fleet-agent-proxy-support]]
+= Use a proxy server with {agent} and {fleet}
+
+Your organization’s security strategy and other considerations may require you
+to use a proxy server between some components in your deployment. For example,
+you may have a firewall rule that prevents endpoints from connecting directly to
+{es}. In this scenario, you can set up the {agent} to connect to a proxy, then
+the proxy can connect to {es} through the firewall.
+
+Support is available in {agent} and {fleet} for connections through HTTP Connect
+(HTTP 1 only) and Socks5 proxy servers.
+
+This guide describes where a proxy server is allowed in your deployment and how
+to configure proxy settings for {agent} and {fleet}. The steps for deploying the
+proxy server itself are beyond the scope of this article.
+
+NOTE: Some environments require users to authenticate with the proxy. There are
+no explicit settings for proxy authentication in {agent} or {fleet}, except the
+ability to pass credentials in the URL or as keys/tokens in headers, as
+described later.
+
+[discrete]
+[[elastic-agent-proxy-config]]
+== Configure {agent} proxy settings
+
+Configure proxy settings for {agent} when it must connect through a proxy server
+to:
+
+* Download artifacts from `artifacts.elastic.co` for subprocesses or binary
+upgrades
+* Send data to {es}
+* Retrieve agent policies from {fleet-server}
+* Retrieve agent policies from {es} (only needed for agents running {fleet-server})
+* Monitor services (not shown)
+
+image::images/agent-proxy-server.png[Image showing connections between {agent}, {fleet-server}, and {es}]
+
+You can configure {agent} proxy settings by specifying:
+
+* <<host-proxy-env-vars>>
+* <<proxy-settings-in-agent-policy>>
+* <<cli-proxy-settings>>
+
+[discrete]
+[[host-proxy-env-vars]]
+=== Host environment variables
+
+Set environment variables on the host to configure default proxy settings.
+The {agent} uses host environment settings by default if no proxy settings are
+specified elsewhere. You can override host proxy settings later when you
+configure the {agent} and {fleet} settings. The following environment variables
+are available on the host:
+
+|===
+|Variable |Description
+
+|`HTTP_PROXY`
+|URL of the proxy server for HTTP traffic. 
+
+|`HTTPS_PROXY`
+|URL of the proxy server for HTTPS traffic.
+
+|`NO_PROXY`
+|IP addresses or domain names that should not use the proxy. Supports patterns.
+|===
+
+The proxy URL can be a complete URL or `host[:port]`, in which case the `http`
+scheme is assumed. An error is returned if the value is a different form.
+
+[discrete]
+[[where-to-set-proxy-env-vars]]
+==== Where to set proxy environment variables
```

Comment:
@michel-laterman Can you take a look at this section about where to set the environment variables? I had to do a bit of digging and want to make sure I got the details right. I didn't include macOS since I don't have the details about that. Maybe we can add that later. I'll make the other changes on Monday because it's late and I'm out of steam.